core/core.go

```diff
@@ -367,6 +367,9 @@ func (n *IpfsNode) SetupOfflineRouting() error {
 	}
 
 	n.Routing = offroute.NewOfflineRouter(n.Repo.Datastore(), n.PrivateKey)
+
+	n.Namesys = namesys.NewNameSystem(n.Routing)
```

Comment:
No, the problem with that is the offline routing system required you to load your private key, which we want to *avoid at all costs* in offline mode. So we only load the routing system with absolutely needed.